src/plugins/intel_gna/src/runtime/cnn.cpp

```diff
@@ -317,8 +317,12 @@ bool is2D(T&& vec) {
 }
 }  // namespace
 
-void CNNMaxPool(intel_dnn_component_t* component, intel_dnn_number_type_t number_type, const bool sumPoolingOverRide) {
-    if (is2D(component->op.maxpool.poolingStrideXY) || is2D(component->op.maxpool.poolingWindowXY)) {
+void CNNMaxPool(intel_dnn_component_t* component,
+                intel_dnn_number_type_t number_type,
+                const bool fused_with_convolution_2d,
```

Comment:
It would be better to add special field(union) in intel_dnn_component_t, for example is_fused_with_conv and the next step should be fused MaxPool with convolution as transformation.

Reply:
This is fix for GNA_SW_FP32 mode, you suggestion would affect current loop of fusing and require changes in "core" implementation of GNA. I think it is too risky for now to introduce it.